infrastructure/src/test/java/org/corfudb/infrastructure/CorfuSequencerServerTest.java

```diff
@@ -0,0 +1,528 @@
+                assertThat(responseStreamAddressSpaceA.getAddressMap())
+                        .isEqualTo(Roaring64NavigableMap.bitmapOf(globalTailA));
+                assertThat(responseStreamAddressSpaceA.getTrimMark()).isEqualTo(Address.NON_ADDRESS);
+            } else {
+                StreamAddressSpace responseStreamAddressSpaceB =
+                        getStreamAddressSpace(uuidToStreamAddressSpacePairMsg.getAddressSpace());
+                assertThat(finalResponse.getPayload().getStreamsAddressResponse().getLogTail()).isEqualTo(globalTailB);
+                assertThat(responseStreamAddressSpaceB.getAddressMap())
+                        .isEqualTo(Roaring64NavigableMap.bitmapOf(globalTailB));
+                assertThat(responseStreamAddressSpaceB.getTrimMark()).isEqualTo(Address.NEVER_READ);
+            }
+        });
+
+
+        // Case 3: Send StreamsAddressRequestMsg.INVALID RPC call.
+        // SequencerServer should send all streams and their address spaces as the response
+        request = getRequestMsg(
+                getBasicHeader(true, true),
+                CorfuMessage.RequestPayloadMsg.newBuilder()
+                        .setStreamsAddressRequest(Sequencer.StreamsAddressRequestMsg.newBuilder()
+                                .setReqType(Sequencer.StreamsAddressRequestMsg.Type.INVALID)
+                                .build())
+                        .build()
+        );
+
+        sequencerServer.handleMessage(request, mockChannelHandlerContext, mockServerRouter);
+
+        verify(mockServerRouter, times(3)).sendResponse(responseCaptor.capture(), any(ChannelHandlerContext.class));
+        response = responseCaptor.getValue();
+        assertTrue(compareBaseHeaderFields(request.getHeader(), response.getHeader()));
+        assertTrue(response.getPayload().hasServerError());
+        assertTrue(response.getPayload().getServerError().hasUnknownError());
+        // A SerializerException is expected.
+        ByteString bs = response.getPayload().getServerError().getUnknownError().getThrowable();
+        try (ObjectInputStream ois = new ObjectInputStream(bs.newInput())) {
+            Throwable payloadThrowable = (Throwable) ois.readObject();
+            assertTrue(payloadThrowable instanceof IllegalArgumentException);
+        } catch (Exception ex) {
+            fail("Missing an IllegalArgumentException from the SequencerServer");
+        }
+    }
+
+    /**
+     * Test that when client sends the SequencerMetricsRequestMsg, the server responds with
+     * SequencerMetricsResponseMsg with SequencerStatus.READY as its content.
+     */
+    @Test
+    public void sequencerMetricsRequest() {
+        sequencerServer = new SequencerServer(mockServerContext, spyFactoryHelper);
+        sequencerServer.setSequencerEpoch(0L);
+        RequestMsg request = getRequestMsg(
+                getBasicHeader(true, true),
```

Comment:
Why does these values differ from the ones set when the client sends the message?

Reply:
Fixed them. Thank you.